content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,6 +36,218 @@ are covered here as well as the `System.Diagnostics` API.
+
+### Dependencies {#example-app-dependencies}
+
+- [.NET SDK](https://dotnet.microsoft.com/download/dotnet) 6+
+
+### Create and launch an HTTP Server
+
+To highlight the difference between instrumenting a _library_ and a standalone
+_app_, split out the dice rolling into a _library file_, which then will be
+imported as a dependency by the _app file_.
+
+Create the _library file_ named `Dice.cs` and add the following code to it:
+
+```csharp
+/*Dice.cs*/
+namespace otel
+{
+    public class Dice
+    {
+        private int min;
+        private int max;
+
+        public Dice(int min, int max)
+        {
+            this.min = min;
+            this.max = max;
+        }
+
+        public List<int> rollTheDice(int rolls)
+        {
+            List<int> results = new List<int>();
+
+            for (int i = 0; i < rolls; i++)
+            {
+                results.Add(rollOnce());
+            }
+            
+            return results;
+        }
+
+        private int rollOnce()
+        {
+            return Random.Shared.Next(min, max + 1);
+        }
+    }
+}
+```
+
+Create the _app file_ `DiceController.cs` and add the following code to it:
+
+```csharp
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
+
+namespace otel
+{
+    public class DiceController : ControllerBase
+    {
+        private ILogger<DiceController> logger;
+
+        public DiceController(ILogger<DiceController> logger)
+        {
+            this.logger = logger;
+        }
+
+        [HttpGet("/rolldice")]
+        public List<int> RollDice(string player, int? rolls)
+        {
+            if(!rolls.HasValue)
+            {
+                logger.LogError("Missing rolls parameter");
+                throw new HttpRequestException("Missing rolls parameter", null, HttpStatusCode.BadRequest);
+            }
+                
+            var result = new Dice(1, 6).rollTheDice(rolls.Value);
+
+            if (string.IsNullOrEmpty(player))
+            {
+                logger.LogInformation("Anonymous player is rolling the dice: {result}", result);
+            }
+            else
+            {
+                logger.LogInformation("{player} is rolling the dice: {result}", player, result);
+            }
+
+            return result;
+        }
+    }
+}
+```
+
+Replace the program.cs content with the following code:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
+To ensure that it is working, run the application with the following command and
+open <http://localhost:8080/rolldice?rolls=12> in your web browser:
+
+```sh
+dotnet build
+dotnet run
+```
+
+You should get a list of 12 numbers in your browser window, for example:
+
+```text
+[5,6,5,3,6,1,2,5,4,4,2,4]
+```
+
+## Manual instrumentation setup
+
+### Dependencies
+
+Install the [OpenTelemetry API and SDK NuGet packages](https://www.nuget.org/profiles/OpenTelemetry):
```

Comment:
https://www.nuget.org/profiles/OpenTelemetry -- this link is not really helpful here. Please point to the specific packages.